guacamole-ext/src/main/java/org/apache/guacamole/token/StandardTokens.java

```diff
@@ -50,6 +50,16 @@
      */
     public static final String CLIENT_ADDRESS_TOKEN = "GUAC_CLIENT_ADDRESS";
 
+    /**
+     * The token that triggers prompts for numeric fields.
+     */
+    public static final String PROMPT_TOKEN_NUMERIC = "-1";
+
+    /**
+     * The token that triggers prompts for string fields.
+     */
+    public static final String PROMPT_TOKEN_STRING = "GUAC_PROMPT";
```

Comment:
In case the whole string needs to be prompt for, maybe you can also check for null values
(same for numeric field as well)
instead of hardcoded -1 or string?

Reply:
If the whole string needs to be prompted, then you just put ```${GUAC_PROMPT}``` in for the whole string.  In Guacamole null or empty values indicate the absence of a parameter and delete that parameter from the underlying database table (in the JDBC modules, anyway), so prompting for an empty value would not be practical and would interfere with some of the basic ways in which parameters are evaluated, aside from which it isn't really necessary.